src/Microsoft.AspNetCore.Server.HttpSys/FeatureContext.cs

```diff
@@ -463,7 +464,15 @@ string IHttpRequestIdentifierFeature.TraceIdentifier
             }
         }
 
-        internal async Task OnStart()
+        bool IHttpMaxRequestBodySizeFeature.IsReadOnly => Request.HasRequestBodyStarted;
+
+        long? IHttpMaxRequestBodySizeFeature.MaxRequestBodySize
+        {
+            get => Request.MaxRequestBodySize;
+            set => Request.MaxRequestBodySize = value;
```

Comment:
It passes strait threw and does throw immediately, you just get a longer call stack.